frontend/public/components/cluster-settings/global-config.tsx

```diff
@@ -41,17 +49,39 @@ class GlobalConfigPage_ extends React.Component<GlobalConfigPageProps, GlobalCon
           return [];
         }).then(items => items.map((i: K8sKind) => ({...i, model})));
     })).then((responses) => {
-      const items = _.sortBy(_.flatten(responses), 'kind', 'asc');
+      const flattenedResponses = _.flatten(responses);
+      const winnowedResponses = flattenedResponses.map(item => ({model: item.model, uid: item.metadata.uid, name: item.metadata.name, namespace: item.metadata.namespace, kind: item.kind}));
+      const sortedResponses = _.sortBy(_.flatten(winnowedResponses), 'kind', 'asc');
+
       this.setState({
-        items,
+        items: sortedResponses,
         loading: false,
       });
     });
   }
 
+  checkFlags(c: plugins.GlobalConfig): GlobalConfigObjectProps {
+    const { flags } = this.props;
+    const { required } = c.properties;
+
+    const requiredArray = required ? _.castArray(required) : [];
+    const requirementMissing = _.some(requiredArray, flag => (
+      flag && (flagPending(flags.get(flag)) || !flags.get(flag))
+    ));
+    if (requirementMissing) {
+      return null;
+    }
+    return c.properties;
+  }
+
   render() {
     const { errorMessage, items, loading } = this.state;
 
+    const globalConfigs = this.getGlobalConfigs();
+    const usableConfigs = globalConfigs.map(item => this.checkFlags(item));
```

Comment:
This can result in an array with `null` entries. I think you want to filter these otherwise it'll end up creating an `<ItemRow` for each `null` entry.

Reply:
Addressed and squashed. Thanks for catching this.